tools/report.collector.in

```diff
@@ -543,45 +543,69 @@ iscfvartrue() {
     getcfvar $1 $2 $3 | egrep -qsi "^(true|y|yes|on|1)"
 }
 
+iscfvarfalse() {
+    getcfvar $1 $2 $3 | egrep -qsi "^(false|n|no|off|0)"
+}
+
 get_logfiles() {
     cf_type=$1
     cf_file="$2"
-    facility_var="logfacility"
 
     case $cf_type in
-	corosync)
+        corosync)
             if [ -f "$cf_file" ]; then
                 debug "Reading $cf_type log settings from $cf_file"
-                if iscfvartrue $cf_type to_syslog "$cf_file"; then
-                    facility_var=syslog_facility
+                # The default value of to_syslog is yes.
+                if ! iscfvarfalse $cf_type to_syslog "$cf_file"; then
+                    facility_cs=$(getcfvar $cf_type syslog_facility "$cf_file")
+                    if [ -z "$facility_cs" ]; then
+                        facility_cs="daemon"
+                    fi
+                    # Always include system logs (if we can find them)
+                    msg="Mark:pcmk:`perl -e 'print time()'`"
+                    logger -p $facility_cs.info $msg >/dev/null 2>&1
+                    sleep 2 # Give syslog time to catch up in case it's busy
+                    findmsg 1 "$msg"
                 fi
+
                 if iscfvartrue $cf_type to_logfile "$cf_file"; then
                     logfile=$(getcfvar $cf_type logfile "$cf_file")
+                    if [ -f "$logfile" ]; then
+                        debug "Log settings found for cluster type $cf_type: $logfile"
+                        echo $logfile
+                    fi
                 fi
             fi
-	    ;;
+            ;;
     esac
 
-    if [ -z "$logfile" ]; then
-        # @TODO Use PCMK_logfile if set
-        logfile="@CRM_LOG_DIR@/pacemaker.log"
-        debug "Log settings not found for cluster type $cf_type, assuming $logfile"
-    fi
-    if [ -f "$logfile" ]; then
-	echo $logfile
-    fi
-
-    if [ "x$facility" = x ]; then
-	facility=`getcfvar $cf_type $facility_var $cf_file`
-	[ "" = "$facility" ] && facility="daemon"
+    facility=$(egrep "^(|export )PCMK_logfacility=" @CONFIGDIR@/pacemaker | tail -n1 | sed -e 's/.*=//')
```

Comment:
Let's functionize this command, and use "grep -E" instead of egrep (which is deprecated, and I believe not POSIX).

e.g. $(get_envvar PCMK_logfacility)

Reply:
Now that I think about it, it probably makes more sense to just source the file and use the variables directly, i.e.

. @CONFIGDIR@/pacemaker